drivers/dma/dma_mcux_lpc.c

```diff
@@ -477,13 +499,24 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 					dst_inc,
 					max_xfer_bytes);
 		} else {
-			/* Enable interrupt and reload for the descriptor
-			 */
-			xfer_config = DMA_CHANNEL_XFER(1UL, 0UL, 1UL, 0UL,
-					width,
-					src_inc,
-					dst_inc,
-					block_config->block_size);
+			if (complete_callback) {
+				/* Enable interrupt and reload for the descriptor
+				 */
+				xfer_config = DMA_CHANNEL_XFER(1UL, 0UL, 1UL, 0UL,
+						width,
+						src_inc,
+						dst_inc,
+						block_config->block_size);
+			} else {
+				/* Disable interrupt, only enable at the end of transfer
+				 * Reload enable.
+				 */
+				xfer_config = DMA_CHANNEL_XFER(1UL, 0UL, 0UL, 0UL,
```

Comment:
instead of if/else, can it just be `DMA_CHANNEL_XFER(1, 0, complete_callback, ...)` ?